From 394108035d350ae662a431c80131f812b5f72dff Mon Sep 17 00:00:00 2001
From: Venture Research <tech@ventureresearch.com>
Date: Fri, 8 Feb 2013 20:22:19 -0600
Subject: [PATCH] lua: update Makefile to use environment build settings

OE-specific parameters, instead of overriding all of these simply use
the ones that are already passed in. Also configure for only Linux...

Signed-off-by: Venture Research <tech@ventureresearch.com>

Updated to work with 3.0.x

Signed-off-by: Armin Kuster <akust808@gmail.com>

---
 deps/lua/src/Makefile | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

Index: redis-3.0.2/deps/lua/src/Makefile
===================================================================
--- redis-3.0.2.orig/deps/lua/src/Makefile
+++ redis-3.0.2/deps/lua/src/Makefile
@@ -5,18 +5,14 @@
 # == CHANGE THE SETTINGS BELOW TO SUIT YOUR ENVIRONMENT =======================
 
 # Your platform. See PLATS for possible values.
-PLAT= none
+PLAT= linux
 
-CC?= gcc
-CFLAGS= -O2 -Wall $(MYCFLAGS)
-AR= ar rcu
-RANLIB= ranlib
-RM= rm -f
-LIBS= -lm $(MYLIBS)
-
-MYCFLAGS=
+MYCFLAGS=-DLUA_USE_LINUX
 MYLDFLAGS=
-MYLIBS=
+MYLIBS=-Wl,-E -ldl -lreadline -lhistory -lncurses
+
+CFLAGS += $(MYCFLAGS)
+LIBS += -lm $(MYLIBS)
 
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
Index: redis-3.0.2/deps/Makefile
===================================================================
--- redis-3.0.2.orig/deps/Makefile
+++ redis-3.0.2/deps/Makefile
@@ -63,7 +63,6 @@ LUA_LDFLAGS+= $(LDFLAGS)
 # lua's Makefile defines AR="ar rcu", which is unusual, and makes it more
 # challenging to cross-compile lua (and redis).  These defines make it easier
 # to fit redis into cross-compilation environments, which typically set AR.
-AR=ar
 ARFLAGS=rcu
 
 lua: .make-prerequisites
Index: redis-3.0.2/deps/lua/Makefile
===================================================================
--- redis-3.0.2.orig/deps/lua/Makefile
+++ redis-3.0.2/deps/lua/Makefile
@@ -33,7 +33,6 @@ INSTALL_DATA= $(INSTALL) -m 0644
 
 # Utilities.
 MKDIR= mkdir -p
-RANLIB= ranlib
 
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
